tools/runfiles/runfiles.rs

```diff
@@ -103,10 +103,25 @@ impl Runfiles {
                 .clone(),
         }
     }
+
+    /// Returns a list of paths to all files in runfiles.
+    ///
+    /// Does not return directories.
+    /// The returned path may not be valid. The caller should check the path's
+    /// validity and that the path exists.
+    pub fn list_files(&self) -> Vec<PathBuf> {
```

Comment:
Would you be willing to add a test for this? And perhaps add a few more sample text files to `//tools/runfiles/data` to make the test a bit more interesting?

Reply:
Hmm I actually would like to see if we could get away with not having this function at all. Other runfiles libraries didn't need it it seems (https://osscs.corp.google.com/bazel/bazel/+/master:tools/cpp/runfiles/runfiles_src.h, https://osscs.corp.google.com/bazel/bazel/+/master:tools/java/runfiles/Runfiles.java, https://osscs.corp.google.com/bazel/bazel/+/master:tools/bash/runfiles/runfiles.bash). 

It seems the main motivation is to be able to find rustfmt manifests in the test data. Is there a reason why we cannot hardcode paths to manifests in the test?

Reply:
Unless a unique binary is compiled for each use of the rustfmt test binary, there isn't really a way to hard code the values. I think it'd be more desirable to continue to use a shared binary and locate manifests in runfiles than to compile unique binaries but if there's a well defined pattern for runfiles libraries then I'd happily respect that. If there is though, I'd question whether or not the implementations for each language could live all in the same place.